sympy/geometry/parabola.py

```diff
@@ -413,3 +418,126 @@ def vertex(self):
             vertex = Point(focus.args[0], focus.args[1] - self.p_parameter)
 
         return vertex
+
+    def area(self, l):
```

Comment:
Would the argument for this be more appropriately a 2D line? It would be more general.

Reply:
Also it isnt' clear to me why we'd want to have an area function for a non-enclosed shape. Would it be better to construct enclosed shapes from 2D elements using boolean operations and then compute areas, centroids, etc?

Reply:
if the line is parallel to directrix then it is ok otherwise we need to find the point of intersection of parabola and line then have to integrate with the proper limits.

Reply:
yes, it is a nice idea to construct enclosed shapes from 2D elements. should I make a separate class for this? 
It will be useful for making cross-section of beams

Reply:
I'm not sure, because I am not familiar with the geometry package. Does it have a built in way to handle boolean operations to generate composite shapes?

Reply:
geometry package does not have a built-in way to handle boolean operations to generate composite shapes.